django/db/models/base.py

```diff
@@ -855,6 +855,7 @@ def save(
             not force_insert
             and deferred_non_generated_fields
             and using == self._state.db
+            and self._is_pk_set()
```

Comment:
It makes no sense to implicitly force an update by setting `update_fields` if the primary is not set.